packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1067,6 +1102,61 @@ class SliverFillViewport extends SliverMultiBoxAdaptorWidget {
   }
 }
 
+class _SliverFractionalPadding extends SingleChildRenderObjectWidget {
+  const _SliverFractionalPadding({
+    this.viewportFraction = 0,
+    Widget sliver,
+  }) : assert(viewportFraction != null),
+       assert(viewportFraction >= 0),
+       assert(viewportFraction <= 0.5),
+       super(child: sliver);
+
+  final double viewportFraction;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) => _RenderSliverFractionalPadding(viewportFraction: viewportFraction);
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSliverFractionalPadding renderObject) {
+    renderObject.setViewportFraction(viewportFraction);
+  }
+}
+
+class _RenderSliverFractionalPadding extends RenderSliverPadding {
+  _RenderSliverFractionalPadding({
+    double viewportFraction = 0,
+  }) : assert(viewportFraction != null),
+       assert(viewportFraction <= 0.5),
+       assert(viewportFraction >= 0),
+       _viewportFraction = viewportFraction,
+       super(padding: EdgeInsets.zero);
+
+  double _viewportFraction;
+  void setViewportFraction(double newValue) {
+    assert(newValue != null);
+    if (_viewportFraction == newValue)
+      return;
+    _viewportFraction = newValue;
+    // Update the padding using the setter. The setter calls marksNeedsLayout internally.
+    padding = padding;
+  }
+
+  @override
+  EdgeInsetsGeometry get padding {
```

Comment:
It seems odd to override the getter for padding like this. Could this just be a private method that gets called whenever we need to set the padding to obtain the value for the padding?

Reply:
Ended up exposing `resolvedPadding` instead of `markNeedsResolution`